### ISSUE TITLE
Exclude patterns in Find are not saved

### ISSUE THREAD

**Issue by stephane-21** [spyder-ide/spyder#15345]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [ ] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [ ] Tried basic troubleshooting (if a bug/error)
    * [x] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [ ] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

"Find in files" advanced options are not saved.

### What steps reproduce the problem?

1. Open "Find in files"
2. Do a search with for example excluding ".venv/*, .git/*"
3. Restart Spyder
4. exclude ".venv/*, .git/*" options have disappeared

### What is the expected output? What do you see instead?
The "exclude" advanced options should be saved.
This is OK with Spyder v4.2.5 .
This does not work with Spyder v5.0.0 .
This does not work with Spyder v5.0.1 .

### Paste Traceback/Error Below (if applicable)

```python-traceback

PASTE TRACEBACK HERE

```

## Versions

* Spyder version: 5.0.1
* Python version: 3.9.0
* Qt version: unknown
* PyQt version: unknown
* Operating System name/version: Xubuntu 20.10 VENV

### Dependencies

```
PASTE DEPENDENCIES HERE
```

**Comment by ccordoba12** [comment]
Hey @stephane-21, thanks for reporting. I can confirm this is a bug and we'll fix it in our next version, to be released next weekend.

**Comment by stephane-21** [comment]
@ccordoba12, I have still some problems with that issue.

I don't know if there is still a bug or if my Spyder 5.0.2 profile is now corrupted (because in the recent past I changed from 5.0.0 to 4.2.5, then from 4.2.5 to 5.0.1, then from 5.0.1 to 4.2.5, then from 4.2.5 to 5.0.2).

My problem is that my exclude pattern in the GUI is `'.git/*, .venv/*, output/*, db_local/*'` but at the moment it is like read-only, any change will disappear in the next session.

If I look at the "spyder.ini" config file (that I never edited manually before), there are these lines :
`
[find_in_files]
enable = True
supported_encodings = ['utf-8', 'iso-8859-1', 'cp1252']
exclude = ['.git/*, .venv/*, output/*, db_local/*', '.git/*, .venv/*, output/*', '*.csv, *.dat, *.log, *.tmp, *.bak, *.orig', '.git/*, .venv/*, *.csv, *.dat, *.log, *.tmp, *.bak, *.orig', '.git/*, .venv/*, db_local/*, *.csv, *.dat, *.log, *.tmp, *.bak, *.orig', '.git/*, .venv/*, db_local/*, output/*, *.csv, *.dat, *.log, *.tmp, *.bak, *.orig']
exclude_regexp = False
search_text_regexp = False
`
All the repetitions are maybe showing a corrupted config file.

Then I edited the config file manually to :
`
exclude = ['.git/*, .venv/*, output/*, db_local/*, *.log']
`
But there is still the "read-only" problem in the GUI.
I think I will reset the Spyder config.

**Comment by ccordoba12** [comment]
> But there is still the "read-only" problem in the GUI.

I can't reproduce this problem, sorry

> I think I will reset my Spyder config.

Yeah, please do and let us know if it fixed your issue

**Comment by stephane-21** [comment]
Today I reset everything with the following procedure and I still have that same bug.

Delete the folder : /home/user/.config/spyder-py3
Delete the folder : /************/my_project/.spyproject
Delete the folder : /************/my_project/.venv/

`cd /************/my_project/`
`python3 -m venv .venv`
`source .venv/bin/activate`
`pip install --upgrade pip`
`pip install spyder==5.0.2`
`deactivate`

I run Spyder with a shortcut script like that :

#!/bin/bash
cd /************/my_project/
source .venv/bin/activate
spyder --new-instance --project=/************/my_project/

Click on : Search > Find in files
By default the exclude pattern is :
`*.csv, *.dat, *.log, *.tmp, *.bak, *.orig`
Search "test"
Replace the default exclude pattern by :
`.venv/*, .git/*`
Search "test"
The search works.
Click on : File > Quit
Run Spyder
The exclude pattern is still :
`*.csv, *.dat, *.log, *.tmp, *.bak, *.orig`

Maybe I am not using Spyder in the right way ?

**Comment by stephane-21** [comment]
Today, I updated to Spyder v5.0.3.
The bug is now a random bug, sometimes the exclude pattern is saved, sometimes not.

I looked at my config file called "spyder.ini".
`exclude = ['.venv/*, .git/*, *.log, output/*, db_local/*, *.txt', '.venv/*, .git/*, *.log, output/*, db_local/*']`
It is a list of 2 strings, which makes me believe that Spyder is aiming to save an individual pattern for each python project.

My main idea is now that :
- The fact that I have several Spyder installations in differents venv.
- The fact that I am launching several Spyder in parallel.
- The fact that all these installation share the same config file.

==> Make Spyder mix up some config settings.

The problem is mainly due to my "dirty" use of Spyder.
==> I should probably use https://direnv.net/.
Is it the only way to launch several Spyder sessions in parallel in a clean way ?
